tutor/utils.py

```diff
@@ -201,7 +202,7 @@ def is_a_tty() -> bool:
 
 
 def execute(*command: str) -> int:
-    click.echo(fmt.command(" ".join(command)))
+    click.echo(fmt.command(shlex.join(command)))
```

Comment:
This is really cool! I did not know about shlex. Unfortunately `shlex.join` was only introduced in Python 3.8: https://docs.python.org/3/library/shlex.html#shlex.join Tutor is supposed to remain compatible with 3.7 until June 2023: https://endoflife.date/python

Luckily, the implementation of `join` is fairly straightforward:

```
def join(split_command):
    """Return a shell-escaped string from *split_command*."""
    return ' '.join(quote(arg) for arg in split_command)
```

This implementation only makes use of [`shlex.quote`](https://docs.python.org/3/library/shlex.html#shlex.quote) which is available since 3.3. Would you mind copying this stub in Tutor with a big reminder to remove it once we deprecate 3.7?

Reply:
@regisb Sure thing! Done.